aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -302,53 +302,42 @@ DimVector TensorIterator::apply_perm_and_mul(IntArrayRef input, int mul) const {
   return res;
 }
 
-void TensorIterator::allocate_outputs() {
+void TensorIterator::allocate_or_resize_outputs() {
   for (int i = 0; i < num_outputs_; i++) {
     auto& op = operands_[i];
-    if (!op.tensor.defined()) {
+    if (!op.tensor.defined() || op.will_resize) {
       TORCH_INTERNAL_ASSERT(op.is_type_defined(), "no type for operand", i);
       int element_size = elementSize(op.target_dtype);
-      auto requires_channels_last_2d_output_ =
-          ndim() == 4 && requires_channels_last_2d_output();
-      auto requires_channels_last_3d_output_ =
-          ndim() == 5 && requires_channels_last_3d_output();
-      if (requires_channels_last_2d_output_ || requires_channels_last_3d_output_) {
-        auto tensor_shape = invert_perm(shape_);
-        op.tensor = at::empty(tensor_shape, op.options());
-        if (requires_channels_last_2d_output_) {
-          op.tensor.unsafeGetTensorImpl()->empty_tensor_restride(
-              MemoryFormat::ChannelsLast);
-        } else {
-          op.tensor.unsafeGetTensorImpl()->empty_tensor_restride(
-              MemoryFormat::ChannelsLast3d);
+      op.stride_bytes = compatible_stride(element_size);
+      // check if permutation is just an inverted order
+      bool inverted = true;
+      for (int i = 1; i <= ndim(); i++) {
+        if (perm_[i - 1] != ndim() - i) {
+          inverted = false;
+          break;
+        }
+      }
+      auto tensor_shape = invert_perm(shape_);
+      if (inverted) {
+        if (!op.tensor.defined())
+        // can just return contiguous output
+        // it is faster because it avoids allocating 0 size tensor and
+        // resizing and restriding it
+          op.tensor = at::empty(tensor_shape, op.options());
```

Comment:
so we might just delete this branch if empty_strided is faster? maybe expand the comment

Reply:
There's still unpermuting of strides that needs to be done even if empty_strided is faster, depending on perf impact we might ignore it and delete this branch.